### PR TITLE
Implement custom configs and refactor

### PR DESCRIPTION
**Pull request by eugenezhernakov**
* Allow app to use custom settings configuration from JSONs speeding up tests
* Add README

### CODE REVIEW

File: example/src/Application/index.tsx [hunk 1]
```diff
@@ -16,7 +15,29 @@ import {
...
+  STORAGE_SETTINGS_KEY,
+} from '../Data/Constants';
+import { Buffer } from 'buffer';
+import { LaunchArguments } from 'react-native-launch-arguments';
+interface MyExpectedArgs {
+  customSettings?: string;
+}
+const args = LaunchArguments.value<MyExpectedArgs>();
+
+const getSettingsFromEnv = () => {
```
Comment (eugenezhernakov) [inline example/src/Application/index.tsx:31]: Does this section need more checks?

File: example/e2e/functional.test.ts [hunk 2]
```diff
@@ -180,6 +192,10 @@ describe('E2E Functional Tests', () => {
   });
 
   it('should return error message upon a failed 3DS2 payment transaction', async () => {
+    await launchApp(defaultConfig);
+    if (await isAndroid()) {
+      await device.disableSynchronization();
+    }
```
Comment (stefan-tudor) [inline example/e2e/functional.test.ts:198]: You have these lines repeated in many places, maybe extracting them as a separate function to save some space and reduce cognitive complexity will be a nice thing to do

File: example/src/Application/index.tsx (hunk 1 above)
Comment (stefan-tudor) [reply example/src/Application/index.tsx:31]: try/catch block will help with catching exceptions when the JSON is malformed

File: example/e2e/processJSONFile.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import fs from 'fs';
+import path from 'path';
+
+const deepMerge = (target: any, source: any) => {
```
Comment (stefan-tudor) [inline example/e2e/processJSONFile.ts:4]: I think you can use lodash here to achieve the same thing with something like [_.merge()](https://lodash.com/docs#merge), we have lodash already listed as a dependency

File: example/e2e/processJSONFile.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  const rawData = fs.readFileSync(filePath, 'utf-8');
+  const jsonData = JSON.parse(rawData);
+
+  deepMerge(jsonData, replacements);
+
+  const jsonString = JSON.stringify(jsonData);
+  const base64Settings = Buffer.from(jsonString).toString('base64');
+
+  return base64Settings;
```
Comment (stefan-tudor) [inline example/e2e/processJSONFile.ts:28]: This stuff may fail if the file is not found, IO issues, etc. Please add a try/catch block and handle the exceptions

**Review by stefan-tudor — APPROVED** [review]
This should speed up a lot the tests, some small things here and there, overall nice stuff!  👍

**Pull request merged**